apps/condo/domains/meter/access/Meter.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Generated by `createschema meter.Meter 'number:Text; billingAccountMeter?:Relationship:BillingAccountMeter:SET_NULL; account?:Relationship:BillingAccount:SET_NULL; property:Relationship:Property:CASCADE; unitName:Text; place?:Text; resource:Relationship:MeterResource:CASCADE;'`
+ */
+
+const { Meter } = require('../utils/serverSchema')
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const { Property } = require('@condo/domains/property/utils/serverSchema')
+const { checkPermissionInUserOrganizationOrRelatedOrganization } = require('@condo/domains/organization/utils/accessSchema')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+const { queryOrganizationEmployeeFromRelatedOrganizationFor, queryOrganizationEmployeeFor } = require('@condo/domains/organization/utils/accessSchema')
+const { Resident: ResidentServerUtils } = require('@condo/domains/resident/utils/serverSchema')
+const { get, uniq, compact } = require('lodash')
+
+
+async function canReadMeters ({ authentication: { item: user }, context }) {
+    if (!user) return throwAuthenticationError()
+    if (user.isAdmin) return {}
+
+    if (user.type === RESIDENT) {
+        const residents = await ResidentServerUtils.getAll(context, { user: { id: user.id } })
+        if (residents.length === 0) {
+            return false
+        }
+        const organizations = compact(residents.map(resident => get(resident, ['organization', 'id'])))
+        if (organizations.length > 0) {
+            return {
+                organization: { id_in: uniq(organizations) },
```

Comment:
it look like a security bug

Reply:
yes, with this login resident has access to all Meters and MeterReadings in organization, fix and write tests for it in f9731f3 and 3f80308